packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -995,6 +995,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         const pendingRuntimeState = context.pendingLocalState as IPendingRuntimeState | undefined;
         const baseSnapshot: ISnapshotTree | undefined = pendingRuntimeState?.baseSnapshot ?? context.baseSnapshot;
 
+        const maxSnapshotCacheDurationMs = this._storage?.policies?.maximumCacheDurationMs;
+        if (maxSnapshotCacheDurationMs !== undefined && maxSnapshotCacheDurationMs > 5 * 24 * 60 * 60 * 1000) {
```

Comment:
Consider moving this to GarbageCollector so we have the GC logic in one place.

Reply:
Hmm maybe. So we'd take in the param still but just for validation?